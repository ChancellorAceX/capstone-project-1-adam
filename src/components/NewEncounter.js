import React from 'react';
import { withRouter } from 'react-router';
import NewCampaignCharacter from './NewCampaignCharacter';
import Header from './Header';
import '../styles/NewEncounter.css';
import SelectBeast from './SelectBeast';
import SelectCampaign from './SelectCampaign';

export default withRouter(class NewEncounter extends React.Component {
  render() {
    return (
      <section className='container'>
        <Header
          pageName='New Encounter'
          history={this.props.history}
        />
        <form className='newEncounter' onSubmit={this.props.saveEncounter}>
          <section className='encounterInfo'>
            <section className='labelFieldGroup'><label htmlFor='ename'>Name:</label>
              <input id='ename' className='encountername' placeholder='Encounter Name' onChange={this.props.setAddEncounter} value={this.props.addEncounter.encountername} required /></section>
            <section className='labelFieldGroup'><label htmlFor='edesc'>Description:</label>
              <textarea id='edesc' className='encounterdesc' placeholder='Encounter Description' onChange={this.props.setAddEncounter} value={this.props.addEncounter.encounterdesc} required /></section>
            <section className='labelFieldGroup'><label htmlFor='ecid'>Campaign:</label>
              <select id='ecid' className='encountercampaignid' placeholder='Associated Campaign' onChange={this.props.setAddEncounter} value={this.props.addEncounter.encountercampaignid}>
                <option disabled value={0}>Select Campaign</option>
                {this.props.campaigns.map(campaign =>
                  <SelectCampaign
                    key={`Campaign ${campaign[0]}`}
                    campaign={campaign}
                  />
                )}
              </select>
            </section>
          </section>
          <section className='addMonster' onChange={this.props.setAddMonster}>
            <select className='type' defaultValue={0}>
              <option disabled value={0}>Select Monster</option>
              {this.props.bestiary.map(monster =>
                <SelectBeast
                  key={`Monster ${monster.bid}`}
                  monster={monster}
                />
              )}
            </select>
            <button type='button' onClick={this.props.addToAddMonsters}>+ Add Monster</button>
          </section>
          <section className='tableBox'>
            <table className='addMonsters'>
              <tbody>
                {this.props.addMonsters.map((monster, i) =>
                  <NewCampaignCharacter
                    key={`NewMonster${i}`}
                    index={i}
                    character={monster}
                  />
                )}
              </tbody>
            </table>
          </section>
          <section className='buttonContainer'>
            <button className='encounterSave' type='submit'>+ Save Encounter</button>
          </section>
        </form>
      </section>
    );
  }
});