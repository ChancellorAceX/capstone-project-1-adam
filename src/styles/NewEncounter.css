form.newEncounter {
  position:fixed;
  top:10vh;
  height:82vh;
  width:100vw;
  overflow:auto;
  background-color: black;
  display:flex;
  flex-direction: column;
  justify-content: flex-start;
  align-items: center;
}

section.encounterInfo {
  display: flex;
  width:100vw;
  flex-direction: column;
  justify-content: flex-start;
  align-items: center;
  padding-bottom:3vh;
  border-bottom:solid 1px #b40a0a;
  margin-bottom: 2vh;
}

section.encounterInfo * {
  width:70vw;
}

section.labelFieldGroup {
  margin-top:10px;
}

section.labelFieldGroup label {
  color:#aaaaaa;
}

section.labelFieldGroup input {
  margin:0;
}

section.labelFieldGroup textarea {
  background-color: #222222;
  color:#aaaaaa;
}

section.labelFieldGroup select {
  background-color: #222222;
  color:#aaaaaa;
}

section.addMonster {
  width:100vw;
  display:flex;
  justify-content: space-evenly;
  align-items: center;
}

section.addMonster select {
  background-color: #222222;
  color: #aaaaaa;
}

section.addMonster button {
  font-weight: 600;
  width:fit-content;
  border-radius:20px;
  background-color: #b40a0a;
  color: #aaaaaa;
  text-shadow:-1px -1px 0 #444444, 1px 1px 0 #444444,1px -1px 0 #444444, -1px 1px 0 #444444;
}

table.addMonsters {
  width:90vw;
  color:#aaaaaa;
}

section.tableBox {
  margin-top:15px;
  width:95vw;
  display:flex;
  flex-direction: column;
  align-items: center;
}

section.buttonContainer {
  position:fixed;
  width:100vw;
  left:0;
  bottom:8vh;
  height:30px;
  display:flex;
  justify-content: center;
  align-items: center;
  border-top:1px solid #aaaaaa;
  background-color: black;
}

button.encounterSave {
  font-weight: 600;
  width:80vw;
  border-radius:20px;
  background-color: #b40a0a;
  color: #aaaaaa;
  text-shadow:-1px -1px 0 #444444, 1px 1px 0 #444444,1px -1px 0 #444444, -1px 1px 0 #444444;
}