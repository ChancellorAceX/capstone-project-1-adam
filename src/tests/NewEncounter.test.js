import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import NewEncounter from '../components/NewEncounter';

describe('NewEncounter component', () => {
const props = {
  "bestiary": [
    "{ac: \"12\", actions: Array(1), bid: 1, burrowspeed: …}",
    "{ac: \"17 (natural armor)\", actions: Array(5), bid: …}",
    "{ac: \"18 (natural armor)\", actions: Array(2), bid: …}",
    "{ac: \"14 (natural armor), 11 while prone\", actions:…}",
    "{ac: \"17 (natural armor, shield)\", actions: Array(1…}",
    "{ac: \"17 (natural armor)\", actions: Array(4), bid: …}",
    "{ac: \"17 (natural armor)\", actions: Array(1), bid: …}",
    "{ac: \"19 (natural armor)\", actions: Array(3), bid: …}",
    "{ac: \"12\", actions: Array(1), bid: 8, burrowspeed: …}",
    "{ac: \"21 (natural armor)\", actions: Array(6), bid: …}"
  ],
  "campaigns": [
    "[1, \"Azeroth\"]",
    "[6, \"Fillory\"]",
    "[7, \"Grishaverse\"]"
  ],
  "addEncounter": {
    "encountername": "",
    "encounterdesc": "",
    "encountercampaignid": 0
  },
  "setAddEncounter": "ƒ () {}",
  "addMonster": {
    "monstername": "Name",
    "initiative": 0,
    "ac": "AC",
    "hp": "",
    "maxhp": "Max HP",
    "pcbestiaryid": "bid"
  },
  "addMonsters": [
    "{ac: \"AC\", hp: \"\", initiative: 0, maxhp: \"Max HP\", …}"
  ],
  "addToAddMonsters": "ƒ () {}",
  "setAddMonster": "ƒ () {}",
  "saveEncounter": "ƒ () {}",
  "history": {
    "length": 20,
    "action": "PUSH",
    "location": "{hash: \"\", key: \"nj3xjv\", pathname: \"/new/encounter…}",
    "createHref": "ƒ createHref() {}",
    "push": "ƒ push() {}",
    "replace": "ƒ replace() {}",
    "go": "ƒ go() {}",
    "goBack": "ƒ goBack() {}",
    "goForward": "ƒ goForward() {}",
    "block": "ƒ block() {}",
    "listen": "ƒ listen() {}"
  },
  "loggedInToggle": "ƒ () {}",
  "location": {
    "pathname": "/new/encounter",
    "search": "",
    "hash": "",
    "key": "nj3xjv"
  },
  "match": {
    "path": "/new/encounter",
    "url": "/new/encounter",
    "isExact": true,
    "params": "{}"
  }
};

  it('renders the full NewEncounter component', () => {
    const wrapper = shallow(<NewEncounter {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});