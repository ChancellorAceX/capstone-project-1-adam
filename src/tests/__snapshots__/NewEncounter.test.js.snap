// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewEncounter component renders the full NewEncounter component 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
